layers.go

```diff
@@ -378,6 +378,48 @@ func (r *layerStore) Load() error {
 	return err
 }
 
+func (r *layerStore) LoadReadOnly() error {
+	rpath := r.layerspath()
+	data, err := ioutil.ReadFile(rpath)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	layers := []*Layer{}
+	idlist := []string{}
+	ids := make(map[string]*Layer)
+	names := make(map[string]*Layer)
+	compressedsums := make(map[digest.Digest][]string)
+	uncompressedsums := make(map[digest.Digest][]string)
+	if err = json.Unmarshal(data, &layers); len(data) == 0 || err == nil {
+		idlist = make([]string, 0, len(layers))
+		for n, layer := range layers {
+			ids[layer.ID] = layers[n]
+			idlist = append(idlist, layer.ID)
+			for _, name := range layer.Names {
+				names[name] = layers[n]
+			}
+			if layer.CompressedDigest != "" {
+				compressedsums[layer.CompressedDigest] = append(compressedsums[layer.CompressedDigest], layer.ID)
+			}
+			if layer.UncompressedDigest != "" {
+				uncompressedsums[layer.UncompressedDigest] = append(uncompressedsums[layer.UncompressedDigest], layer.ID)
+			}
+			if layer.MountLabel != "" {
+				label.ReserveLabel(layer.MountLabel)
+			}
+			layer.ReadOnly = true
+		}
+		err = nil
+	}
+	r.layers = layers
+	r.idindex = truncindex.NewTruncIndex(idlist)
+	r.byid = ids
+	r.byname = names
+	r.bycompressedsum = compressedsums
+	r.byuncompressedsum = uncompressedsums
+	return err
```

Comment:
I would appreciate a closer look to this function @nalind. I basically tried to isolate the read-only operations from `Load()`.

Reply:
would it make sense to refactor the two functions into one?  There are not many differences between `Load` and `LoadReadOnly`